pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java

```diff
@@ -420,19 +423,13 @@ void testMessageExpiryWithTimestampNonRecoverableException() throws Exception {
         when(mock.getLastPosition()).thenReturn(PositionImpl.EARLIEST);
 
         PersistentMessageExpiryMonitor monitor = new PersistentMessageExpiryMonitor(mock, c1.getName(), c1, null);
-        Position previousMarkDelete = null;
-        for (int i = 0; i < totalEntries; i++) {
-            monitor.expireMessages(1);
```

Comment:
Just need to call `monitor.expireMessages` once and then use Awaitility to assert markDeletePositiion.